src/hotspot/os/linux/cgroupV2Subsystem_linux.cpp

```diff
@@ -180,6 +180,16 @@ jlong CgroupV2Subsystem::memory_and_swap_limit_in_bytes() {
   return swap_limit;
 }
 
+jlong CgroupV2Subsystem::memory_and_swap_usage_in_bytes() {
+    jlong memory_usage = memory_usage_in_bytes();
+    if (memory_usage >= 0) {
+        char* mem_swp_current_str = mem_swp_current_val();
+        jlong swap_current = limit_from_str(mem_swp_current_str);
+        return memory_usage + swap_current;
```

Comment:
`limit_from_str()` might return negative values in case of `OS_CONTAINER_ERROR` (`-2`) or `max` => `-1`. Please account for that. For example:
```
return memory_usage + (swap_current >= 0 ? swap_current : 0);
```

Reply:
Makes sense. I adjusted the coding - btw should we add some UL logging in the case of failures/negative values ?

Reply:
Up to you. It's probably not worth it.